src/Schema/AbstractAsset.php

```diff
@@ -108,14 +148,46 @@ protected function _setName(string $name): void
                 return;
             }
         } else {
-            $identifiers = [];
+            $parsedName = null;
         }
 
-        switch (count($identifiers)) {
-            case 0:
-                $this->identifiers = [];
+        try {
+            $this->setName($parsedName);
+        } catch (Throwable $e) {
```

Comment:
Should this be narrowed down to something more precise, like `NotImplemented`?

Reply:
It can also throw `InvalidName`. Since this is just an upgrade path, that doesn't add any new functionality, the idea is to prefer safety over purity. No matter what happens in `setName()`, I'd rather it to result in a false-positive deprecation message than an uncaught exception.